src/main/scala/rocket/PMP.scala

```diff
@@ -161,6 +161,25 @@ class PMPChecker(lgMaxSize: Int)(implicit p: Parameters) extends CoreModule()(p)
     val hit = pmp.hit(io.addr, io.size, lgMaxSize, prevPMP)
     val ignore = default && !pmp.cfg.l
     val aligned = pmp.aligned(io.addr, io.size, lgMaxSize, prevPMP)
+
+    cover(!default && pmp.cfg.a === 0x0, "The cfg access is set to no access",    "Cover PMP access mode setting")
+    cover(!default && pmp.cfg.a === 0x1, "The cfg access is set to TOR access",   "Cover PMP access mode setting")
+    cover(!default && pmp.cfg.a === 0x2, "The cfg access is set to NA4 access",   "Cover PMP access mode setting")
+    cover(!default && pmp.cfg.a === 0x3, "The cfg access is set to NAPOT access", "Cover PMP access mode setting")
+    cover(!default && pmp.cfg.l === 0x1, "The cfg lock is set to high",           "Cover PMP lock mode setting")
+    
+    // Not including Write and no Read permission as the combination is reserved
+    cover(!default && (Cat(pmp.cfg.x, pmp.cfg.w, pmp.cfg.r) === 0x0), "The permission is set to no access",                        "Cover PMP access permission setting") 
+    cover(!default && (Cat(pmp.cfg.x, pmp.cfg.w, pmp.cfg.r) === 0x1), "The permission is set to Read only access",                 "Cover PMP access permission setting") 
+    cover(!default && (Cat(pmp.cfg.x, pmp.cfg.w, pmp.cfg.r) === 0x3), "The permission is set to Read and Write only access",       "Cover PMP access permission setting") 
+    cover(!default && (Cat(pmp.cfg.x, pmp.cfg.w, pmp.cfg.r) === 0x4), "The permission is set to Execution only access",            "Cover PMP access permission setting") 
+    cover(!default && (Cat(pmp.cfg.x, pmp.cfg.w, pmp.cfg.r) === 0x5), "The permission is set to Read and Execution only access",   "Cover PMP access permission setting") 
+    cover(!default && (Cat(pmp.cfg.x, pmp.cfg.w, pmp.cfg.r) === 0x7), "The permission is set to Read, Write and Execution access", "Cover PMP access permission setting")
```

Comment:
You can loopify this but discard the W-not-R cases with something like

    for ((name, idx) <- Seq("no", "RO", "", "RW", "X", "RX", "", "RWX").zipWithIndex; if name.nonEmpty)